source/templates.tex

```diff
@@ -1898,25 +1892,68 @@
 Two \grammarterm{template-id}{s} refer to the same
 class, function, or variable if
 \begin{itemize}
-\item {their \grammarterm{template-name}{s},
-\grammarterm{operator-function-id}{s}, or \grammarterm{literal-operator-id}{s}
-refer to the same template and}
-\item {their corresponding type \grammarterm{template-argument}{s} are the
-same type and}
-\item {their corresponding non-type \grammarterm{template-argument}{s} of
-pointer-to-member type refer to the same class member or are both the null member
-pointer value and}
-\item {their corresponding non-type \grammarterm{template-argument}{s} of
-reference type refer to the same object or function and}
-\item {their remaining corresponding
-non-type \grammarterm{template-argument}{s}
-have the same type and value
-after conversion to the type of the \grammarterm{template-parameter},
-where they are considered to have the same value if they compare equal
-with the \tcode{==} operator\iref{expr.eq}, and}
-\item {their corresponding template \grammarterm{template-argument}{s} refer
-to the same template.}
+\item
+their \grammarterm{template-name}{s},
+\grammarterm{operator-function-id}{s}, or
+\grammarterm{literal-operator-id}{s}
+refer to the same template, and
+
+\item
+their corresponding type \grammarterm{template-argument}{s}
+are the same type, and
+
+\item
+their corresponding non-type \grammarterm{template-argument}{s}
+are template-argument-equivalent (see below)
+after conversion to the type of the \grammarterm{template-parameter}, and
+
+\item
+their corresponding template \grammarterm{template-argument}{s}
+refer to the same template.
+\end{itemize}
+
+\pnum
+Two values are \defn{template-argument-equivalent} if
+they are of the same type and
+\begin{itemize}
+\item
+they are of integral type and their values are the same, or
+
+\item
+they are of floating-point type and their values are identical, or
+
+\item
+they are of type \tcode{std::nullptr_t}, or
+
+\item
+they are of enumeration type and their values are the same,%
+\footnote{The identity of enumerators is not preserved.} or
+
+\item
+they are of pointer type and they have the same pointer value, or
+
+\item
+they are of pointer-to-member type and they refer to the same class member
+or are both the null member pointer value, or
+
+\item
+they are of reference type and they refer to the same object or function, or
+
+\item
+they are of array type and their corresponding elements are template-argument-equivalent,%
+\footnote{An array as a \grammarterm{template-parameter} decays to a pointer.} or
+
+\item
+they are of union type and either
+they both have no active member or
+they have the same active member and their active members are template-argument-equivalent, or
+
+\item
+they are of class type and
+their corresponding direct subobjects and reference members are template-argument-equivalent.
 \end{itemize}
```

Comment:
Same here. The long items call for two-space indentation.

That said, the separation of `\item`s by blank lines looks pretty nice, so maybe it's okay. Maybe a new entry to https://github.com/cplusplus/draft/wiki/Specification-Style-Guidelines#latex-source-code-formatting is needed. I think this style is novel, as I don't remember seeing it before.

Reply:
The format is actually in accordance to the [new guidelines](https://github.com/cplusplus/draft/issues/3127#issuecomment-549577948), they're just not in the wiki yet. Disregard my review.